timemachine/fe/free_energy.py

```diff
@@ -1243,6 +1274,12 @@ def assert_ensembles_compatible(state_a: InitialState, state_b: InitialState):
         # also, assert barostat and integrator are self-consistent
         assert intg_a.temperature == baro_a.temperature
 
+        # The protein and water parameters should match exactly for the water sampling parameters
```

Comment:
Is this necessary for correctness, or is the assertion that the water parameters are consistent in the all pairs and interaction group potentials sufficient?